lib/instrumentation/core/http-outbound.js

```diff
@@ -58,6 +59,18 @@ function getDefaultHostName(opts) {
   return opts.hostname || opts.host || DEFAULT_HOST
 }
 
+function extractHostPortViaProxy(opts) {
```

Comment:
Please document this function.

Reply:
I'm also not seeing how this is solving your need.  pathname is coming from `url.parse` which always is a URI and will never have the protocol in the path.

Reply:
Protocol will be present on requests via proxy like:
`proxyRequest.js`
```
require "newrelic";
const axios = require('axios');

axios.get('https://google.com/index.html').then(res => {
  console.info(res.data);
});
```
`HTTPS_PROXY=http://proxy:8080 HTTP_PROXY=http://proxy:8080 NO_PROXY=localhost,.internal node proxyRequest.js`